### PR TITLE
ci(linter): mandatory scope for PR and commit messages

### PR DESCRIPTION
**Pull request by b00f**
## Description

This PR mandates specific rules for the scope in PR titles:

1. The scope must be present, for example, `feat(grpc): sign transaction using wallet client`.
2. The scope must not include uppercase letters.
3. The scope must include one of the following names: `github, daemon, cmd, gtk, shell, wallet, committee, config, consensus, crypto, docs, execution, genesis, network, node, sandbox, scripts, sortition, state, store, sync, main, txpool, types, util, version, grpc, http, jsonrpc, nanomsg`.
4. Multiple scopes are allowed, for example, `feat(grpc, wallet): sign transaction using wallet client`.

### CODE REVIEW

**Comment by kehiy** [comment]
@b00f For `docs` scope sometimes we have `docs:` style. it would be `docs(docs):` ? or we need to put `docs(daemon)` for example?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,51 +1,63 @@
...
 
 - Each commit should represent a single, atomic change to the codebase.
   Avoid making multiple unrelated changes in a single commit.
-- Commit message should not be longer than 50 characters.
-- Commit message should start with lowercase and not ends with punctuation.
-- Write commit message in the imperative: "fix bug" and not "fixed bug" or "fixes bug".
-- Use the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) format for commit messages and Pull Request titles.
+- Use the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) format for commit messages and
+  Pull Request titles.
+
```
Comment (kehiy) [inline CONTRIBUTING.md:59]: Do we need to provide supported scopes here?

Comment (b00f) [reply CONTRIBUTING.md:58]: These scopes may change in the future, so we provided a link to the linter (source of scopes).

**Comment by b00f** [comment]
For `docs` type we can use `other` as scope:  `docs(other)`

**Review by kehiy — APPROVED** [review]
Looks good to me.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kehiy) [reply CONTRIBUTING.md:58]: Got it. Thanks.

**Pull request merged**